File: Structural/Facade/cpp/test_facade.cpp
#include "facade.hpp"
#include <gtest/gtest.h>

using namespace facade_pattern;

/**
 * @brief Tests the Facade pattern to ensure the interface simplifies interaction with subsystems.
 */
TEST(FacadePatternTest, SimplifiedInterface) {
	// Create the Facade instance
	Facade facade;

	// Execute the Facade operation
	std::string result = facade.operation();

	// Expected output
	std::string expected =
		"Facade: Coordinating subsystems:\n"
		"Subsystem1: Operation 1\n"
		"Subsystem2: Operation 2\n"
		"Subsystem3: Operation 3\n";

	// Verify the result
	EXPECT_EQ(result, expected);
}

File: Structural/Facade/python/test_facade.py
"""
Unit tests for the Facade pattern.
"""

import unittest
from facade import Facade


class TestFacade(unittest.TestCase):
	"""
	Tests the Facade class to ensure it simplifies interaction with subsystems.
	"""

	def test_facade_operation(self):
		"""
		Test the Facade's ability to coordinate subsystems.
		"""
		# Create the Facade instance
		facade = Facade()

		# Execute the Facade operation
		result = facade.operation()

		# Expected output
		expected = (
			"Facade: Coordinating subsystems:\n"
			"Subsystem1: Operation 1\n"
			"Subsystem2: Operation 2\n"
			"Subsystem3: Operation 3\n"
		)

		# Verify the result
		self.assertEqual(result, expected)


if __name__ == "__main__":
	unittest.main()
